### DISCUSSION TITLE
Compile using Native AOT if possible

### DISCUSSION THREAD

**Discussion by alphajoza** [vpnhood/VpnHood#173] [Ideas]
Since you've switched to .NET 7 in v2.4.319 (prerelease), if possible ([Native AOT limitations](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/#limitations-of-native-aot-deployment)), compile the server binary to native code for decreasing startup time and memory footprint.

**Comment by trudyhood** [comment]
As you see the new version (prerelease) is compiled as win-x64 & linux-x64
Is that what you were looking for?

**Reply by alphajoza** [reply]
What is currently available in the prerelease assets is a self-contained package. What I mean is what languages such as Rust and Golang produce, i.e., a single, native, binary.

[Per documentation](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/#publish-native-aot---cli), this could be done, via adding the following property into project file:

```
<PropertyGroup>
    <PublishAot>true</PublishAot>
</PropertyGroup>
```

**Reply by trudyhood** [reply]
Thank you for sharing that. I understood that it mainly improves the startup, while the VpnHood server is a long job in the background. Also, there is some limitation and stability consideration. Have you compiled it with AOT and found significant improvement?
